support/globalconfig/network.go

```diff
@@ -31,4 +31,16 @@ func ReconcileNetworkConfig(cfg *configv1.Network, hcp *hyperv1.HostedControlPla
 		cfg.Spec.ExternalIP = globalConfig.Network.Spec.ExternalIP
 		cfg.Spec.ServiceNodePortRange = globalConfig.Network.Spec.ServiceNodePortRange
 	}
+
+	// Without this, the CNOs proxy controller refuses to reconcile the proxy status
+	// The CNO only populates this after MTU probing and it is required for the proxy
+	// controller to populate proxy status. Proxy not being populated makes the MTU
+	// probing fail if there is a proxy. Get out of the deadlock by initially populating
+	// it if unset.
```

Comment:
can we add an if proxyEnabled check to the ifs?

Reply:
Updated. I don't think it matters since we only set this if it is unset but it won't hurt either.

Reply:
> it matters since we only set this if it is unset but it won't hurt either.

Thanks, my point is, without the check, for the non proxy scenario, we'd intercepting the operator behaviour, we wouldn't be giving the operator the chance to set status based on the state of the world as designed (only populates this after MTU probing), we'd be faking status. We are faking it any way for the proxy scenario but it's justified to fix de deadlock issue you described.